aea/mail/base.py

```diff
@@ -477,9 +485,19 @@ def _stop(self):
             ).result()
             self._send_loop_task.cancel()
 
-        if self._loop.is_running():
+        if self._connect_all_task is not None:
+            self._connect_all_task.cancel()
+        if self._disconnect_all_task is not None:
+            self._disconnect_all_task.cancel()
+
+        for connection in [c for c in self.connections
+                           if c.connection_status.is_connected or c.connection_status.is_connecting]:
+            asyncio.run_coroutine_threadsafe(connection.disconnect(), self._loop).result()
```

Comment:
this part will be executed when `multiplexer.disconnect` is called - and in turn, the `self._stop()`. 
We force the stop of the connection that are connected or connecting.